tt_metal/programming_examples/matmul_multicore_reuse_mcast/matmul_multicore_reuse_mcast.cpp

```diff
@@ -11,100 +11,42 @@
 #include "tt_metal/detail/tt_metal.hpp"
 #include "tt_metal/programming_examples/matmul_common/bmm_op.hpp"
 #include <algorithm>
+#include "tt_metal/common/tilize_untilize.hpp"
 
 using namespace tt::constants;
 using namespace std;
 using namespace tt;
 using namespace tt::tt_metal;
 
 
-template <typename T>
-std::vector<T> tilize(std::vector<T> data, int rows, int cols) {
-    TT_ASSERT(rows % 32 == 0);
-    TT_ASSERT(cols % 32 == 0);
-    int num_tiles_r = rows / 32;
-    int num_tiles_c = cols / 32;
-    std::vector<T> result;
-    for(auto r = 0; r < num_tiles_r; r++) {
-        for(auto c = 0; c < num_tiles_c; c++) {
-            for(auto j = 0; j < 32; j++) { // tile rows
-                for(auto i = 0; i < 32; i++) { // tile cols
-                    // each row of tiles is 32x32 * num_tiles_c
-                    // each row within the row of tiles is cols
-                    // each col of tiles is 32
-                    // pick row of tiles, pick the row within the tile, pick col tile
-                    int index = r * 32 * 32 * num_tiles_c + j * cols + c * 32 + i;
-                    result.push_back(data.at(index));
-                }
-            }
-        }
-    }
-    return convert_to_tile_layout(result);
-}
-
-
-template <typename T>
-std::vector<T> untilize(std::vector<T> data, int rows, int cols) {
-    int TileWidth = 32;
-    int TileHeight = 32;
-    TT_ASSERT(rows % TileHeight == 0);
-    TT_ASSERT(cols % TileWidth == 0);
-    int elements_in_tile = TileHeight*TileWidth;
-    int num_tiles_r = rows / TileHeight;
-    int num_tiles_c = cols / TileWidth;
-    std::vector<T> result;
-    for(auto r = 0; r < num_tiles_r; r++) {
-        for(auto i = 0; i < TileHeight; i++) {
-            for(auto c = 0; c < num_tiles_c; c++) {
-                int offset = r * elements_in_tile * num_tiles_c + c * elements_in_tile + i * TileWidth;
-                for(auto j = 0; j < TileWidth; j++) {
-                    result.push_back(data.at(offset + j));
-                }
-            }
-        }
-    }
-    return convert_to_flat_layout(result);
-}
-
-
-void golden_matmul(vector<bfloat16> a, vector<bfloat16> b, vector<uint32_t>& output,
+void golden_matmul(vector<bfloat16> a, vector<bfloat16> b, vector<bfloat16>& output,
```

Comment:
Nitpick: vectors 'a' and 'b' should be passed by reference, not by value.

Reply:
done